api/peering_test.go

```diff
@@ -2,14 +2,155 @@ package api
 
 import (
 	"context"
+	"reflect"
 	"testing"
 	"time"
 
+	"github.com/hashicorp/serf/testutil/retry"
 	"github.com/stretchr/testify/require"
 
 	"github.com/hashicorp/consul/sdk/testutil"
 )
 
+func peerExistsInPeerListings(peer *Peering, peerings []*Peering) bool {
+	for _, aPeer := range peerings {
+		isEqual := (peer.PeerID == aPeer.PeerID) &&
+			(reflect.DeepEqual(peer.PeerCAPems, aPeer.PeerCAPems)) &&
+			(peer.PeerServerName == aPeer.PeerServerName) &&
+			(peer.Partition == aPeer.Partition) &&
+			(peer.Name == aPeer.Name) &&
+			(reflect.DeepEqual(peer.PeerServerAddresses, aPeer.PeerServerAddresses)) &&
+			(peer.State == aPeer.State) &&
+			(peer.CreateIndex == aPeer.CreateIndex) &&
+			(peer.ModifyIndex) == aPeer.ModifyIndex
+
+		if isEqual {
+			return true
+		}
+	}
+
+	return false
+}
+
+func TestAPI_Peering_Read(t *testing.T) {
+	t.Parallel()
+	c, s := makeClientWithCA(t)
+	defer s.Stop()
+	s.WaitForSerfCheck(t)
+	ctx := context.Background()
+	peerings := c.Peerings()
+
+	t.Run("call Read with no name", func(t *testing.T) {
+		resp, qm, err := peerings.Read(ctx, PeeringRequest{}, nil)
+
+		// basic checks
+		require.EqualError(t, err, "peering name cannot be empty")
+		require.Empty(t, qm)
+		require.Empty(t, resp)
+	})
+
+	t.Run("read token that does not exist on server", func(t *testing.T) {
+		resp, qm, err := peerings.Read(ctx, PeeringRequest{Name: "peer1"}, nil)
+
+		// basic checks
+		require.NotNil(t, err) // 404
+		require.Empty(t, qm)
+		require.Empty(t, resp)
+	})
+
+}
+
+// TestAPI_Peering_List
+func TestAPI_Peering_List(t *testing.T) {
```

Comment:
This test has subtests that mutate a shared state; we should either call `c, s := makeClientWithCA(t)` once in every test or stop using `t.Run`